activerecord/lib/active_record/insert_all.rb

```diff
@@ -32,10 +32,9 @@ def updatable_columns
     end
 
     def primary_keys
-      Array(model.primary_key)
+      Array(connection.schema_cache.primary_keys(model.table_name))
```

Comment:
```suggestion
      connection.schema_cache.primary_keys(model.table_name)
```

`primary_keys` should return an Array already, iirc.

Reply:
Looks like having `schema_cache` return an array for `primary_keys` causes issues because array return values get suppressed.

https://github.com/rails/rails/blob/1aa90ddc1c2ae5d44f64f41c6eda3345054d69c2/activerecord/lib/active_record/attribute_methods/primary_key.rb#L92-L93

https://github.com/rails/rails/blob/1aa90ddc1c2ae5d44f64f41c6eda3345054d69c2/activerecord/lib/active_record/attribute_methods/primary_key.rb#L122-L129